### PR TITLE
chore: add error types that should be retried

### PR DESCRIPTION
**Pull request by yikuaibro**

### CODE REVIEW

File: internal/log.go [hunk 1]
```diff
@@ -19,6 +22,20 @@ func (l *logger) Printf(ctx context.Context, format string, v ...interface{}) {
...
+
+func (l *logger) Fatalf(ctx context.Context, format string, v ...interface{}) {
+	_ = l.log.Output(2, fmt.Sprintf(format, v...))
+	os.Exit(1)
+}
+
+func (l *logger) Fatalln(ctx context.Context, v ...interface{}) {
+	_ = l.log.Output(2, fmt.Sprintln(v...))
+	os.Exit(1)
+}
```
Comment (vmihailenco) [inline internal/log.go:37]: Changes to the logging are unrelated and should be in a separate PR.

File: error.go [hunk 2]
```diff
@@ -55,6 +55,15 @@ func shouldRetry(err error, retryTimeout bool) bool {
...
 	}
+	if s == "Connection reset by peer" {
+		return true
+	}
+	if strings.HasPrefix(s, "ERR redis temporary failure") {
+		return true
+	}
+	if strings.HasPrefix(s, "FLUSHDB ") {
+		return true
+	}
```
Comment (vmihailenco) [inline error.go:66]: Please add a brief comment / reasoning why these errors must be retried.

Comment (yikuaibro) [reply error.go:66]: ok, I will add a comment / reasoning, thank you.

File: internal/log.go (hunk 1 above)
Comment (yikuaibro) [reply internal/log.go:37]: Understand.
